src/rules/arrowReturnShorthandRule.ts

```diff
@@ -60,10 +60,10 @@ interface Options {
 
 function walk(ctx: Lint.WalkContext<Options>): void {
     const { sourceFile, options: { multiline } } = ctx;
-    ts.forEachChild(sourceFile, function cb(node: ts.Node): void {
+    return ts.forEachChild(sourceFile, function cb(node: ts.Node): void {
```

Comment:
Time measurement isn't necessary if you think it means the difference between tail call and not tail call. If `return` made it into a tail call, then this code would not get a stack overflow:
```js
function f(x) {
    if (x === 0)
        return console.log("done");
    else
        return f(x - 1);
}
f(30000);
```